From 42691dc1c7ba8954132b56f876c2294e1e8c6e21 Mon Sep 17 00:00:00 2001
From: Xiang Wang <xiang.a.wang@intel.com>
Date: Thu, 12 Mar 2015 00:04:21 +0800
Subject: [PATCH] i2c: designware: remove delay for shared host

In https://android.intel.com/#/c/327239/, a small
delay is added for shared host. But this delay
brings performance issues to boot time and display
turn on. So remove this delay after verifying there
is no impact to i2c sharing with PUNIT.

Change-Id: Icda60e77726a12cf1d42a68245bdc317273aa072
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-6891
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1195
Signed-off-by: Xiang Wang <xiang.a.wang@intel.com>
---
 drivers/i2c/busses/i2c-designware-core.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/i2c/busses/i2c-designware-core.c b/drivers/i2c/busses/i2c-designware-core.c
index 93146cd..b588656 100644
--- a/drivers/i2c/busses/i2c-designware-core.c
+++ b/drivers/i2c/busses/i2c-designware-core.c
@@ -703,9 +703,6 @@ i2c_dw_xfer(struct i2c_adapter *adap, struct i2c_msg msgs[], int num)
 	dev_dbg(dev->dev, "%s: msgs: %d\n", __func__, num);
 
 	mutex_lock(&dev->lock);
-	if (dev->shared_host)
-		usleep_range(10000, 12000);
-
 	pm_runtime_get_sync(dev->dev);
 
 	reinit_completion(&dev->cmd_complete);
-- 
1.9.1

